src/base/Roles/CrossChain/CrossChainTellerBase.sol

```diff
@@ -0,0 +1,204 @@
+        external
+        requiresAuth
+    {
+        if (messageGasLimit == 0) {
+            revert CrossChainTellerBase_ZeroMessageGasLimit();
+        }
+        Chain storage chain = selectorToChains[chainSelector];
+        chain.allowMessagesTo = true;
+        chain.targetTeller = targetTeller;
+        chain.messageGasLimit = messageGasLimit;
+
+        emit ChainAllowMessagesTo(chainSelector, targetTeller);
+    }
+
+    /**
+     * @notice Stop messages from a chain.
+     * @dev Callable by MULTISIG_ROLE.
+     */
+    function stopMessagesFromChain(uint32 chainSelector) external requiresAuth {
+        Chain storage chain = selectorToChains[chainSelector];
+        chain.allowMessagesFrom = false;
+
+        emit ChainStopMessagesFrom(chainSelector);
+    }
+
+    /**
+     * @notice Stop messages to a chain.
+     * @dev Callable by MULTISIG_ROLE.
+     */
+    function stopMessagesToChain(uint32 chainSelector) external requiresAuth {
+        Chain storage chain = selectorToChains[chainSelector];
+        chain.allowMessagesTo = false;
+
+        emit ChainStopMessagesTo(chainSelector);
+    }
+
+    /**
+     * @notice Set the gas limit for messages to a chain.
+     * @dev Callable by OWNER_ROLE.
+     */
+    function setChainGasLimit(uint32 chainSelector, uint64 messageGasLimit) external requiresAuth {
+        if (messageGasLimit == 0) {
+            revert CrossChainTellerBase_ZeroMessageGasLimit();
+        }
+        Chain storage chain = selectorToChains[chainSelector];
+        chain.messageGasLimit = messageGasLimit;
+
+        emit ChainSetGasLimit(chainSelector, messageGasLimit);
+    }
+
+    
+    /**
+     * @dev function to deposit into the vault AND bridge crosschain in 1 call
+     * @param depositAsset ERC20 to deposit
+     * @param depositAmount amount of deposit asset to deposit
+     * @param minimumMint minimum required shares to receive
+     * @param data Bridge Data
+     */
+    function depositAndBridge(ERC20 depositAsset, uint256 depositAmount, uint256 minimumMint, BridgeData calldata data) external payable{
+        uint shareAmount = _erc20Deposit(depositAsset, depositAmount, minimumMint, msg.sender);
+        bridge(shareAmount, data);
+    }
+
+
+    /**
+     * @notice Preview fee required to bridge shares in a given feeToken.
+     */
+    function previewFee(uint256 shareAmount, BridgeData calldata data)
+        external
+        view
+        returns (uint256 fee)
+    {
+        return _quote(shareAmount, data);
+    }
+
+    /**
+     * @dev bridging code to be done without deposit, for users who already have vault tokens
+     * @param shareAmount to bridge
+     * @param data bridge data
+     */
+    function bridge(uint256 shareAmount, BridgeData calldata data) public payable returns(bytes32 messageId){
+        _beforeBridge(shareAmount, data);
+        messageId = _bridge(shareAmount, data);
+        _afterBridge(shareAmount, data, messageId);
+    }
```

Comment:
Personally I find it misleading that `_beforeBridge` and `_afterBridge` are two seprate functions when it's not supposed to be overridden. 

The style suggests that `_beforeBridge` and `_afterBridge` are hooks that can be modified, but for this contract, the `_beforeBridge` is not meant to be a virtual function. 

imo keeping only the overriddable hooks to be a separate internal function and putting the `_beforeBridge` logic into the `bridge` function itself is less misleading.

Reply:
I'll move _beforeBridge() logic to bridge()

For this though it is something that will affect PR1 as well... I'll talk to you directly about this